internal/js_parser/js_parser.go

```diff
@@ -13891,11 +13891,14 @@ func (p *parser) visitFn(fn *js_ast.Fn, scopeLoc logger.Loc) {
 		}
 	}
 
-	if fn.Name != nil {
+	if !isExpr && fn.Name != nil {
 		p.recordDeclaredSymbol(fn.Name.Ref)
 	}
 
 	p.pushScopeForVisitPass(js_ast.ScopeFunctionArgs, scopeLoc)
+	if fn.Name != nil && isExpr {
```

Comment:
I think this is correct scope, but let me know if I should move it to function body instead.

Reply:
Added a test to verify that it is indeed the correct scope.